packages/runtime/container-runtime/src/pendingStateManager.ts

```diff
@@ -509,6 +545,10 @@ export class PendingStateManager implements IDisposable {
 					pendingMessage.batchIdContext.clientId,
 					pendingMessage.batchIdContext.batchStartCsn,
 				);
+			// Ignore emptyBatch messages when resubmitting
+			if (pendingMessage.opMetadata?.emptyBatch === true) {
+				continue;
```

Comment:
We also need to do something like this, otherwise the batchId will get lost:

```suggestion
				this.stateHandler.reSubmitBatch([], batchId);
				continue;
```

It's tough to test this now since my PR to check batchId isn't in yet.  I'll make a note of this test case though.

Reply:
I mean, you can write a unit test to show that if there's an empty batch before, there's also one after, I think that's worth doing.

But there's no end-to-end test that will break yet if you get it wrong.